ext/cl_khr_extended_bit_ops.asciidoc

```diff
@@ -0,0 +1,143 @@
+// Copyright 2018-2021 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+[[cl_khr_extended_bit_ops]]
+== Extended Bit Operations
+
+This extension adds OpenCL C functions for performing extended bit operations.
+Specifically, the following functions are added:
+
+* bitfield insert: insert bits from one source operand into another source operand.
+* bitfield extract: extract bits from a source operand, with sign- or zero-extension.
+* bit reverse: reverse the bits of a source operand.
+
+=== General Information
+
+==== Name Strings
+
+`cl_khr_extended_bit_ops`
+
+==== Version History
+
+[cols="1,1,3",options="header",]
+|====
+| *Date*     | *Version* | *Description*
+| 2021-04-22 | 1.0.0     | Initial version.
+|====
+
+==== Dependencies
+
+This extension is written against the OpenCL 3.0 C Language Specification and the OpenCL SPIR-V Environment Specification Version V3.0.6.
+
+This extension requires OpenCL 1.0.
+
+=== New OpenCL C Functions
+
+[source]
+----
+gentype bitfield_insert( gentype base, gentype insert, uint offset, uint count )
+igentype bitfield_extract_signed( gentype base, uint offset, uint count )
+ugentype bitfield_extract_unsigned( gentype base, uint offset, uint count )
+gentype bit_reverse( gentype base )
+----
+
+=== Modifications to the OpenCL C Specification
+
+==== Modify Section 6.15.3. Integer Functions:
+
+Add a new Section 6.15.3.X. Extended Bit Operations: ::
++
+--
+The functions described in the following table can be used with built-in scalar or vector integer types to perform extended bit operations.
+The functions that operate on vector types operate component-wise.
+The description is per-component.
+
+In the table below, the generic type name `gentype` refers to the built-in integer types `char`, `char__n__`, `uchar`, `uchar__n__`, `short`, `short__n__`, `ushort`, `ushort__n__`, `int`, `int__n__`, `uint`, `uint__n__`, `long`, `long__n__`, `ulong`, and `ulong__n__`.
+The generic type name `igentype` refers to the built-in signed integer types `char`, `char__n__`, `short`, `short__n__`, `int`, `int__n__`, `long`, and `long__n__`.
+The generic type name `ugentype` refers to the built-in unsigned integer types `uchar`, `uchar__n__`, `ushort`, `ushort__n__`, `uint`, `uint__n__`, `ulong`, and `ulong__n__`.
+_n_ is 2, 3, 4, 8, or 16.
+
```

Comment:
We may want to mention the interaction with `cles_khr_int64` on embedded profile devices.

Reply:
How important is this?

Looking through the rest of the extensions spec we seem to be fairly dilligent about qualifying `double` and `half` support, but we generally don't qualify `long` support.  Options are: fix them all in a subsequent PR or qualify usage of the `long` type here at least, even though it's not qualified elsewhere.

I'd prefer to fix this in a subsequent PR.

Reply:
A subsequent PR works for me.